src/coreclr/scripts/superpmi-setup.py

```diff
@@ -0,0 +1,395 @@
+#!/usr/bin/env python3
+#
+## Licensed to the .NET Foundation under one or more agreements.
+## The .NET Foundation licenses this file to you under the MIT license.
+#
+##
+# Title               : superpmi-setup.py
+#
+# Notes:
+#
+# Script to setup directory structure to perform SuperPMI collection in CI.
+#
+################################################################################
+################################################################################
+
+
+import subprocess
+import argparse
+
+from os import listdir, path, walk
+from os.path import isfile, join, getsize
+from coreclr_arguments import *
+
+# Start of parser object creation.
+
+parser = argparse.ArgumentParser(description="description")
+
+parser.add_argument("-source_directory", help="path to source directory")
+parser.add_argument("-core_root_directory", help="path to core_root directory")
+parser.add_argument("-arch", help="Architecture")
+parser.add_argument("-mch_file_tag", help="Tag to be used to mch files")
+parser.add_argument("-libraries_directory", help="directory containing assemblies for which superpmi collection to "
+                                                  "be done")
+parser.add_argument("-tests_directory", help="path to managed test artifacts directory")
+parser.add_argument("-max_size", help="Max size of each partition in MB")
+is_windows = platform.system() == "Windows"
+native_binaries_to_ignore = [
```

Comment:
Presumably there needs to be a Linux and macOS version of this list

Reply:
Yes, I have added it in my TODO list of #42783 .